elastic/kibana
# [Dashboard] Fix missing state on short URL alias match redirect

## Summary
Fixes an issue where URL state from short URLs could be lost on an alias match redirect.

**Before**
![before](https://github.com/elastic/kibana/assets/14276393/3d8ace9f-aa96-4243-a925-32d35ea6ccc5)

**After**
![After](https://github.com/elastic/kibana/assets/14276393/90c320b4-ff36-4cc3-924d-86a145f9a142)


> **Note**
> In the after gif - and in any testing, you'll notice that the fixed version shows an error in the Lens Embeddable. This is because the short URL contains a `panels` key with an un-migrated version of the by value Lens panel on the Dashboard. This would not have happened if I had used a by reference panel, and newer versions of Kibana have warnings in place to prevent the saving of un-migrated state to a short URL in this way.

To test:
1. Download and extract this ES data which has two Dashboards in two spaces (with a URL conflict) and has a short URL pointing to the conflicting Dashboard URL with a filter attached.
[esData_short_url_alias_redirect.zip](https://github.com/elastic/kibana/files/12316691/esData_short_url_alias_redirect.zip)

2. Start ES pointing to where you extraced that data `yarn es snapshot -E path.data=path/to/data`
3. Try navigating to the short URL `{Your_local_url}/s/new-space/goto/efbe7e20-37ae-11ee-8876-5107f2fdfac9`
4. The Dashboard should load properly, with the filter set.

### Checklist

Delete any items that are not applicable to this PR.
